src/main/java/uk/co/ramp/event/ContactEventProcessor.java

```diff
@@ -66,27 +67,52 @@ private boolean isContactIsolated(ContactEvent contact, Case caseA, Case caseB,
 
-    Optional<InfectionEvent> evaluateExposures(ContactEvent c, int time) {
-        Case personA = getMostSevere(population.get(c.to()), population.get(c.from()));
-        Case personB = personA == population.get(c.to()) ? population.get(c.from()) : population.get(c.to());
-
-        boolean dangerMix = personA.isInfectious() && personB.virusStatus() == SUSCEPTIBLE;
-
-        if (dangerMix && distributionSampler.uniformBetweenZeroAndOne() < c.weight() / diseaseProperties.exposureTuning()) {
-            LOGGER.debug("       DANGER MIX");
-
-            InfectionEvent infectionEvent = ImmutableInfectionEvent.builder().
-                    id(personB.id()).time(c.time() + 1).
-                    oldStatus(SUSCEPTIBLE).nextStatus(EXPOSED).
-                    exposedTime(time).exposedBy(personA.id()).
-                    build();
-
-
-            return Optional.of(infectionEvent);
-
-        }
-        return Optional.empty();
-    }
+	/**
+	 * Randomly generate an exposure event, based on contact weight between two persons.
+	 * 
+	 * For contact weight C >= 0, probability of exposure is given by
+	 * P(exposure=1|C) = Sigmoid(bias + exponent * ln(C))
+	 *                 = ( exp(bias) * C^{exponent} ) / (1 + exp(bias) * C^{exponent}),
+	 *                 
+	 * where exponent > 0 and Sigmoid(x) = 1 / (1 + exp(-x)).
+	 * 
+	 * As long as exponent>0, this formula satisfies P(exposure=1|C=0)==0, i.e.,
+	 * contact weight == 0, which corresponds to no contact, is guaranteed to unchange the status.
+	 * This is a necessary boundary condition, because lack of contact record is implicitly assumed to be contact weight == 0.
+	 * 
+	 * The bias and exponent parameters control level of the probability and
+	 * sensitivity of the probability to the contact weight, respectively.
+	 *  
+	 * For an intuitive parameter specification in {@link DiseaseProperties},
+	 * instead of specifying bias in [-infty, infty], 
+	 * probability U == P(exposure=1|C=1) = {@link exposureProbability4UnitContact#DiseaseProperties},
+	 * whose support is (0, 1), should be provided.
+	 * Then it is automatically converted into exp(bias) = U/(1-U) based on Sigmoid(bias) = U.
+	 * 
+	 * @param c
+	 * @param time
+	 * @return
+	 */
+	Optional<InfectionEvent> evaluateExposures(ContactEvent c, int time) {
+		final Case personA = getMostSevere(population.get(c.to()), population.get(c.from()));
+		final Case personB = personA == population.get(c.to()) ? population.get(c.from()) : population.get(c.to());
+
+		final boolean dangerMix = personA.isInfectious() && personB.virusStatus() == SUSCEPTIBLE;
+		
+		final double expBias = diseaseProperties.exposureProbability4UnitContact()/(1.0-diseaseProperties.exposureProbability4UnitContact());
+		final double exposureProb = 1. / (1. + 1./(expBias*FastMath.pow(c.weight(), diseaseProperties.exposureExponent())));
+
+		if (dangerMix&& distributionSampler.uniformBetweenZeroAndOne() < exposureProb) {
+			LOGGER.debug("       DANGER MIX");
```

Comment:
Probably best if @sibylleM can review the math here

Reply:
@vinopm @maxlogutility Looks good to me.